haystack/nodes/summarizer/transformers.py

```diff
@@ -182,35 +159,21 @@ def predict(self, documents: List[Document], generate_single_summary: Optional[b
 
         result: List[Document] = []
 
-        if generate_single_summary:
-            for context, summarized_answer in zip(contexts, summaries):
-                cur_doc = Document(content=summarized_answer["summary_text"], meta={"context": context})
-                result.append(cur_doc)
-        else:
-            for context, summarized_answer, document in zip(contexts, summaries, documents):
-                cur_doc = Document(content=summarized_answer["summary_text"], meta=document.meta)
-                cur_doc.meta.update({"context": context})
-                result.append(cur_doc)
+        for summary, document in zip(summaries, documents):
+            document.meta.update({"summary": summary["summary_text"]})
+            result.append(document)
 
         return result
 
     def predict_batch(
-        self,
-        documents: Union[List[Document], List[List[Document]]],
-        generate_single_summary: Optional[bool] = None,
-        batch_size: Optional[int] = None,
+        self, documents: Union[List[Document], List[List[Document]]], batch_size: Optional[int] = None
```

Comment:
I agree that it would simplify things...
But I see that most of the batch methods in nodes accept `Union[List[Document], List[List[Document]]]`
Can you help me to better understand this point?

Reply:
I'll dig into this and be back with some explanations :smile: I honestly don't know right now why we would ever expect `List[Document]` here

Reply:
Soo this issue hints to the fact that this is indeed redundant: https://github.com/deepset-ai/haystack/issues/2529

So I believe for now we should keep this PR as it is and open another issue about simplifying this signature.